.github/workflows/windows.yml

```
@@ -413,52 +413,129 @@ jobs:
           cmake --build bld --config '${{ matrix.type }}' --target test-ci
 
   msvc:
-    name: 'msvc (${{ matrix.arch }}, ${{ matrix.plat }}, ${{ matrix.config }})'
+    name: 'msvc (${{ matrix.arch }}, ${{ matrix.plat }}, ${{ matrix.name }})'
     runs-on: windows-latest
     timeout-minutes: 30
     strategy:
       matrix:
         include:
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=ON  -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=ON ', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=OFF', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON          -DENABLE_UNICODE=OFF', type: 'Debug' }
+          # name should be unique and without special chars. It creates a folder with that name for the cache.
+        - name: 'schannel'
+          install: 'zstd zlib nghttp2 libssh2[core,zlib]'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
+          config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=ON  -DCURL_USE_SCHANNEL=ON  -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON'
+        - name: 'openssl'
+          install: 'zstd zlib nghttp2 nghttp3 openssl libssh2'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
+          config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DBUILD_SHARED_LIBS=OFF -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON'
+        - name: 'no ssl'
+          install: 'zstd zlib'
+          arch: 'x64'
+          plat: 'windows'
+          type: 'Debug'
+          tflags: '~1516 ~2301 ~2302 ~2303 ~2307'
+          config: '-DENABLE_DEBUG=ON -DENABLE_UNICODE=OFF -DCURL_USE_SCHANNEL=OFF -DCURL_ZSTD=ON -DHTTP_ONLY=ON'
       fail-fast: false
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: 'restore cache dependencies'
+        uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4
+        with:
+          path: 'cache/${{ matrix.name }}'
+          key: ${{ matrix.plat }}-${{ matrix.name }}
+          restore-keys: ${{ matrix.plat }}-${{ matrix.name }}
+
+      - name: 'vcpkg init'
+        timeout-minutes: 5      
```

```suggestion
        timeout-minutes: 5
```
Delete line ending spaces here.

(I wish GitHub would make these visible.)

---

Can you make a script that check that?

---

Can you make a scrip for that?

---

`checksrc` could probably do that. With the right amount of Perl-fu.

---

Nah, checksrc checks C code, having it checking yaml seems like an overstep and a job for a separate script.

---

Yep, it's indeed usually not C code that is hit. I have a script doing it locally, but written in a foreign language:
https://github.com/vszakats/hb/blob/83173b807b4eceda0a46704aa6544e92240cc21c/bin/commit.hb#L579-L656

---

we can have python script that sit inside script job that check yaml format.

---

We could reasonably go for _all_ files (not just YAML) with some exclusions (e.g. test definitions, certs), and allowing tabs in some make files.

Sticking with Perl would make sense to avoid adding an extra build dependency. Also some of the necessary logic is already implemented in `checksrc`. That could be reused for a whitespace checker.

---

The checker landed in 1ccdad64efecbf3167d6f60737037157011c1468 #14031.

---

I will rebase it to here.
This PR is count as a bugfix? that it can be merge to curl in feature freeze time?
Because I don't change the curl at all ....

---

Yes, I thnk it's fine to merge CI changes outside the feature window.

What new test cases are you thinking of?

---

> What new test cases are you thinking of?

When I added here the other libraries, your tests cases was activate and check more cases on msvc jobs. that what I meant.
But since you ask, I am planning to add more working libraries from vcpkg. and add wup case.
But this will be next PR. as soon this one will be merge.